vars/named_vars.go

```diff
@@ -0,0 +1,47 @@
+package vars
+
+import (
+	"fmt"
+	"strings"
+)
+
+type NamedVariables map[string]Variables
+
+// Get checks var_source if presents, then forward var to underlying secret manager.
+// A `varDef.Name` with a var_source looks like "myvault:foo", where "myvault" is
+// the var_source name, and "foo" is the real var name that should be forwarded
+// to the underlying secret manager.
+func (m NamedVariables) Get(varDef VariableDefinition) (interface{}, bool, error) {
+	var sourceName, varName string
+	parts := strings.Split(varDef.Name, ":")
+	if len(parts) == 1 {
+		// No source name, then no need to query named vars.
+		return nil, false, nil
+	} else if len(parts) == 2 {
+		sourceName = parts[0]
+		varName = parts[1]
+	} else {
+		return nil, false, fmt.Errorf("invalid var: %s", varDef.Name)
+	}
+
+	if vars, ok := m[sourceName]; ok {
+		return vars.Get(VariableDefinition{Name: varName})
+	}
+
+	return nil, false, nil
```

Comment:
Make sense.

Reply:
@vito As I'm working on `input_vars` today, and I rethink this over. The problem I'm facing is, `input_vars` are not secrets, if we merge `input_vars`'s `NamedVars` into `var_sources`'s `NamedVars`, then `input_vars` would be redacted, which I think is not intended behavior.

For your concern, since `var_sources` are not allowed to have duplicate names, as long as we make sure `input_vars` don't use duplicate names as `var_sources`, then problem will not happen. In the future, for project, we may also check `var_sources` within a project to not have duplicate names. WDYT?

As 5.8.0 is not release yet, if you agree, now is the best time to change it back to return no error.

Reply:
Sorry, I don't follow - how does returning an error here impact this decision?

Redacting `input_vars` *could* be what the user wants - it really depends on the contents of the vars. Someone could have generated a username/password to a temporary account, for example. Maybe the decision to redact should be made fine-grained, i.e. an attribute of the returned var, and things like `input_vars` would allow you to configure it?

Reply:
Returning an error here make a step to have only one `NamedVars`. `input_vars` has to be merged into `var_sources`. If here doesn't return error, then we may have multiple `NamedVars`, one for `var_sources` and the other for `input_vars`.

As `input_vars` is a simplified alternative of input files, and content of input files are not redacted, so I'm thinking maybe not redact `input_vars`.

But I don't have strong opinion on that. Let's keep the current way here to return an error.

Reply:
Ah, I see. Let's leave it as-is for now and hash this detail out in the `input_vars` RFC (...once I have time to take a look at it :sweat:).